clients/src/test/java/org/apache/kafka/clients/producer/internals/RecordAccumulatorTest.java

```diff
@@ -784,6 +786,40 @@ public void testIdempotenceWithOldMagic() throws InterruptedException {
         accum.append(tp1, 0L, key, value, Record.EMPTY_HEADERS, null, 0);
     }
 
+    @Test
+    public void testRecordHeadersWithV1() {
```

Comment:
i would be happier if we have an integration test (with an embedded broker booted) to make sure this works. there may be code after the records builder that barfs